project1/src/main/java/uk/co/cpsd/javaproject1/Goat.java

```diff
@@ -1,53 +1,178 @@
-            energyLevel-=2;
-            lastEnergyDecreaseTick=currentTick;
-            return energyLevel <= 0;
         }
+    }
 
-        System.out.println("Goat " + animalId + " energy decreased to " + energyLevel);
-        return false;
+    @Override
+    public DecisionInfo animalDecisionMaking(World world) {
+
+        Map<Point, List<Object>> scanedNeighbourHoodByGoat = world.scanNeighbour(getX(), getY());
+
+        // 1. Priority: Eat if hungry
+        if (isHungry()) {
+            for (Map.Entry<Point, List<Object>> entry : scanedNeighbourHoodByGoat.entrySet()) {
+                if (entry.getValue().contains("grass")) {
+                    return new DecisionInfo(DecisionType.EAT, entry.getKey());
+                }
+            }
+        }
+
+        // 2. Priority: Reproduce (check nearby goats)
+        for (Map.Entry<Point, List<Object>> entry : scanedNeighbourHoodByGoat.entrySet()) {
+            for (Object obj : entry.getValue()) {
+                if (obj instanceof Goat otherGoat && this.canReproduceWith(otherGoat, world.getTotalTicks())) {
+                    return new DecisionInfo(DecisionType.REPRODUCE, entry.getKey());
+                }
+            }
+        }
+
+        // 3. Priority: Flee from danger
+        Point safe = findRandomSafePos(scanedNeighbourHoodByGoat);
+        if (!safe.equals(new Point(getX(), getY()))) {
+            return new DecisionInfo(DecisionType.FLEE, safe);
+        }
+
+        // 4. Default: Random move
+        Point randomMove = findRandomPos(scanedNeighbourHoodByGoat);
+        return new DecisionInfo(DecisionType.WANDER, randomMove);
+    }
+
+    public Point findRandomSafePos(Map<Point, List<Object>> neighbourHoodPos) {
+        List<Point> safeTiles = neighbourHoodPos.entrySet()
+                .stream()
+                .filter(e -> !e.getValue().contains("lion"))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList());
+
+        if (!safeTiles.isEmpty()) {
+            return safeTiles.get(new Random().nextInt(safeTiles.size()));
+        }
+
+        // All tiles are dangerous stay in its place
+        return new Point(this.getX(), this.getY());
+    }
+
+    public Point findRandomPos(Map<Point, List<Object>> neighbourHoodPos) {
+        List<Point> allTiles = new ArrayList<>(neighbourHoodPos.keySet());
+        if (!allTiles.isEmpty()) {
+            return allTiles.get(new Random().nextInt(allTiles.size()));
+        }
+        return new Point(this.getX(), this.getY());
     }
+
+    public boolean canReproduceWith(Goat otherGoat, int currentTick) {
+        if (otherGoat == this)
+            return false;
+
+        boolean oppositeGender = this.getGender() != otherGoat.getGender();
+        boolean pairsHaveEenergy = this.energyLevel >= 10 && otherGoat.energyLevel >= 10;
+        boolean sinceLastReproduce = currentTick - this.lastReproductionTick >= 5
```

Comment:
Again, not sure there's anything goat-specific here.  And maybe an "isFertile" method? (whose period may differ for male and female goats?)